api/token.go

```diff
@@ -259,3 +286,14 @@ func (a *API) clearCookieToken(ctx context.Context, w http.ResponseWriter) {
 		Path:     "/",
 	})
 }
+
+func getExpiry(tokenPayload string) int64 {
+	jsonString, _ := base64.RawStdEncoding.DecodeString(tokenPayload)
+	var payload map[string]interface{}
+	err := json.Unmarshal(jsonString, &payload)
+	if err != nil {
+		logrus.WithError(err).Warn("Failed to parse expiry from cached token - disabling cache")
+		return 0
+	}
+	return int64(payload["exp"].(float64))
```

Comment:
I am assuming it is guaranteed that the payload will have exp field.

Reply:
Yes `exp` is always there.